<?xml version="1.0"?>
<robot name="space_robot">
  
  
  <link name="base_link">
    <visual>
      <geometry>
        <cylinder length="0.43" radius="0.15"/>
      </geometry>
      <origin xyz="0 0 0.215" rpy="0 0 0"/>
    </visual>
    <inertial>
      <origin xyz="0.015 0.03 0.15" rpy="0 0 0"/>
      <mass value="15"/>
      <inertia ixx="0.04" ixy="0.0" ixz="0.0" iyy="0.04" iyz="0.0" izz="0.02"/>
    </inertial>
  </link>



  <link name="left_arm">
    <visual>
      <geometry>
        <box size="0.2 .02 .06"/>
      </geometry>
      <origin xyz="0.1 0 0" rpy="0 0 0"/>
    </visual>
    <inertial>
      <origin xyz="0.1 0 0" rpy="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="0.004" ixy="0.0" ixz="0.0" iyy="0.004" iyz="0.0" izz="0.002"/>
    </inertial>
  </link>

  <link name="left_forearm">
    <visual>
      <geometry>
        <box size="0.2 .02 .06"/>
      </geometry>
      <origin xyz="0.1 0 0" rpy="0 0 0"/>
    </visual>
    <inertial>
      <origin xyz="0.1 0 0" rpy="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="0.004" ixy="0.0" ixz="0.0" iyy="0.004" iyz="0.0" izz="0.002"/>
    </inertial>
  </link>

  <link name="left_grip">
    <visual>
      <geometry>
        <box size="0.02 .08 .02"/>
      </geometry>
      <origin xyz="0.01 0 0" rpy="0 0 0"/>
    </visual>
    <inertial>
      <origin xyz="0.01 0 0" rpy="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="0.004" ixy="0.0" ixz="0.0" iyy="0.004" iyz="0.0" izz="0.002"/>
    </inertial>
  </link>  


  
  <link name="right_arm">
    <visual>
      <geometry>
        <box size="0.2 .02 .06"/>
      </geometry>
      <origin xyz="0.1 0 0" rpy="0 0 0"/>
    </visual>
    <inertial>
      <origin xyz="0.1 0 0" rpy="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="0.004" ixy="0.0" ixz="0.0" iyy="0.004" iyz="0.0" izz="0.002"/>
    </inertial>
  </link>

  <link name="right_forearm">
    <visual>
      <geometry>
        <box size="0.2 .02 .06"/>
      </geometry>
      <origin xyz="0.1 0 0" rpy="0 0 0"/>
    </visual>
    <inertial>
      <origin xyz="0.1 0 0" rpy="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="0.004" ixy="0.0" ixz="0.0" iyy="0.004" iyz="0.0" izz="0.002"/>
    </inertial>
  </link>

<link name="right_grip">
    <visual>
      <geometry>
        <box size="0.02 .08 .02"/>
      </geometry>
      <origin xyz="0.01 0 0" rpy="0 0 0"/>
    </visual>
    <inertial>
      <origin xyz="0.01 0 0" rpy="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="0.004" ixy="0.0" ixz="0.0" iyy="0.004" iyz="0.0" izz="0.002"/>
    </inertial>
  </link>  
  



  <joint name="right_shoulder" type="revolute">
    <parent link="base_link"/>
    <child link="right_arm"/>
    <origin xyz="0.15 0.1 0.1" rpy="0 0 0"/>
    <axis xyz="0 0 1" /> 
    <limit effort="1" lower="-1.57" upper="2.57" velocity="0.5"/>
  </joint>

  <joint name="right_elbow" type="revolute">
    <parent link="right_arm"/>
    <child link="right_forearm"/>
    <origin xyz="0.2 0 0" rpy="0 0 0"/>
    <axis xyz="0 0 1" />
    <limit effort="1" lower="-2.54" upper="2.54" velocity="0.5"/>
  </joint>

  <joint name="right_wrist" type="revolute">
    <parent link="right_forearm"/>
    <child link="right_grip"/>
    <origin xyz="0.2 0 0" rpy="0 0 0"/>
    <axis xyz="1 0 0" />
    <limit effort="1" lower="-2.54" upper="2.54" velocity="0.5"/>
  </joint>
  
  

  <joint name="left_shoulder" type="revolute">
    <parent link="base_link"/>
    <child link="left_arm"/>
    <origin xyz="0.15 -0.1 0.1" rpy="0 0 0"/>
    <axis xyz="0 0 1" />
    <limit effort="1" lower="-2.57" upper="1.57" velocity="0.5"/>
  </joint>

  <joint name="left_elbow" type="revolute">
    <parent link="left_arm"/>
    <child link="left_forearm"/>
    <origin xyz="0.2 0 0" rpy="0 0 0"/>
    <axis xyz="0 0 1" />
    <limit effort="1" lower="-2.54" upper="2.54" velocity="0.5"/>
  </joint>

  <joint name="left_wrist" type="revolute">
    <parent link="left_forearm"/>
    <child link="left_grip"/>
    <origin xyz="0.2 0 0" rpy="0 0 0"/>
    <axis xyz="1 0 0" />
    <limit effort="1" lower="-2.54" upper="2.54" velocity="0.5"/>
  </joint>

</robot>